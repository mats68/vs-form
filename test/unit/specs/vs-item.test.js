import Vue from 'vue'
import { shallow, mount } from 'vue-test-utils'

import VsForm from 'src/components/vs-form'
import VsItem from 'src/components/vs-item'

import { Schema, examples } from 'vs-schema'

import Vuetify from 'vuetify'
Vue.use(Vuetify)

// todo Vue.use
Vue.component('vs-form', VsForm)
Vue.component('vs-item', VsItem)

describe('VsItem miniSchema', () => {
  let Cmp, CmpName, CmpRoot

  beforeEach(() => {
    Cmp = shallow(VsItem, {
      propsData: {
        schema: Schema(examples.miniSchema),
        node: 'name'
      }
    })

    CmpName = mount(VsItem, {
      propsData: {
        schema: Schema(examples.miniSchema),
        node: 'name'
      }
    })

    CmpRoot = mount(VsItem, {
      propsData: {
        schema: Schema(examples.miniSchema),
        node: 'root'
      }
    })

  })

  it('shallow render', () => {
    expect(Cmp.element).toMatchSnapshot()
  })

  it('deep render name component', () => {
    expect(CmpName.element).toMatchSnapshot()
  })

  it('deep render root component', () => {
    expect(CmpRoot.element).toMatchSnapshot()
  })

  it('is a vsitem component', () => {
    expect(CmpName.is(VsItem)).toBe(true)
  })
})
