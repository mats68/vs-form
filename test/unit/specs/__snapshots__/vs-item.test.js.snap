// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VsItem miniSchema deep render name component 1`] = `
<div
  class="input-group input-group--dirty input-group--text-field primary--text"
>
  <label
    for="name"
  >
    Name
  </label>
  <div
    class="input-group__input"
  >
    <input
      id="name"
      tabindex="0"
      type="text"
    />
  </div>
  <div
    class="input-group__details"
  >
    <!---->
  </div>
</div>
`;

exports[`VsItem miniSchema deep render root component 1`] = `
<div>
  <p>
    Mini
  </p>
   
  <div
    class="grid-container grid-row"
    options="[object Object]"
  >
    <div
      class="md-4 xs-12 rows-2"
    >
      <div
        class="input-group input-group--dirty input-group--text-field primary--text"
      >
        <label
          for="name"
        >
          Name
        </label>
        <div
          class="input-group__input"
        >
          <input
            id="name"
            tabindex="0"
            type="text"
          />
        </div>
        <div
          class="input-group__details"
        >
          <!---->
        </div>
      </div>
    </div>
    <div
      class="xl-6 xs-12"
    >
      <div
        class="input-group input-group--text-field primary--text"
      >
        <label
          for="vorname"
        >
          Vorname
        </label>
        <div
          class="input-group__input"
        >
          <input
            id="vorname"
            tabindex="0"
            type="text"
          />
        </div>
        <div
          class="input-group__details"
        >
          <!---->
        </div>
      </div>
    </div>
    <div
      class="xl-6 xs-12"
    >
      <div
        class="input-group input-group--text-field primary--text"
      >
        <label
          for="strasse"
        >
          Strasse
        </label>
        <div
          class="input-group__input"
        >
          <input
            id="strasse"
            tabindex="0"
            type="text"
          />
        </div>
        <div
          class="input-group__details"
        >
          <!---->
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`VsItem miniSchema shallow render 1`] = `
<div
  class="input-group input-group--dirty input-group--text-field primary--text"
>
  <label
    for="name"
  >
    Name
  </label>
  <div
    class="input-group__input"
  >
    <input
      id="name"
      tabindex="0"
      type="text"
    />
  </div>
  <div
    class="input-group__details"
  >
    <!---->
  </div>
</div>
`;
